controllers/external/tracker.go

```diff
@@ -47,6 +49,10 @@ func (o *ObjectTracker) Watch(log logr.Logger, obj client.Object, handler handle
 		return nil
 	}
 
+	if o.Cache == nil || o.Scheme == nil {
```

Comment:
Added a check here, so it continue to work as no-op when controller is not set and make sure to check these two are set when controller is set.